packages/template-retail-react-app/app/pages/product-list/partials/page-designer-promotional-banner.test.js

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2024, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import React from 'react'
+import {rest} from 'msw'
+import {screen} from '@testing-library/react'
+import {renderWithProviders} from '@salesforce/retail-react-app/app/utils/test-utils'
+import {mockImageWithText} from '@salesforce/retail-react-app/app/mocks/page-designer'
+import PageDesignerPromotionalBanner from '@salesforce/retail-react-app/app/pages/product-list/partials/page-designer-promotional-banner'
+
+describe('PageDesignerPromotionalBanner', function () {
+    beforeEach(() => {
+        global.server.use(
+            rest.get('*/pages/instagram-promo-banner-desktop', (req, res, ctx) => {
+                return res(ctx.delay(0), ctx.status(200), ctx.json(mockImageWithText))
+            })
+        )
+        global.server.use(
+            rest.get('*/pages/instagram-promo-banner-mobile', (req, res, ctx) => {
+                return res(ctx.delay(0), ctx.status(200), ctx.json(mockImageWithText))
+            })
+        )
+        window.history.pushState({}, 'ProductList', '/uk/en-GB/category/mens-clothing-jackets')
+    })
+
+    test('should render mobile and desktop banner', async () => {
+        renderWithProviders(<PageDesignerPromotionalBanner />)
+        // NOTE: Both mobile and desktop views exist, but they're hidden
+        // with display:none style depending on the browser's view
+        expect(await screen.findByTestId('sf-promo-banner-desktop')).toBeInTheDocument()
+        expect(await screen.findByTestId('sf-promo-banner-mobile')).toBeInTheDocument()
+    })
+
+    test.each([
+        [500, mockImageWithText],
+        [500, undefined]
+    ])('should not render when /page API returns a 500', async (statusCode, response) => {
```

Comment:
🤔 I wonder why the server returns 500 error. According to the API doc, it's supposed to be 404: https://developer.salesforce.com/docs/commerce/commerce-api/references/shopper-experience?meta=getPage

Reply:
Good question... asking the page designer team! https://salesforce-internal.slack.com/archives/C01RW0121HR/p1727793546564629

Reply:
Looks like it was a bug on the SCAPI side. They are looking to resolve it. I'll merge this PR for now and add the test case for the 404 in a separate PR once it's resolved